packages/multi-select-combo-box/src/vaadin-multi-select-combo-box-internal.js

```diff
@@ -140,6 +157,39 @@ class MultiSelectComboBoxInternal extends ComboBoxDataProviderMixin(ComboBoxMixi
     this._toggleElement = this.querySelector('.toggle-button');
   }
 
+  /**
+   * Override combo-box method to group selected
+   * items at the top of the overlay.
+   *
+   * @protected
+   * @override
+   */
+  _setDropdownItems(items) {
+    if (this.readonly || !this.groupSelectedItems) {
+      this._dropdownItems = items;
+      return;
+    }
+
+    if (this.topGroup) {
+      const filteredTopItems = this.topGroup.filter((selectedItem) => {
+        return items.some((item) => this._getItemLabel(item) === this._getItemLabel(selectedItem));
+      });
+
+      const filteredItems = (items || []).filter((item) => {
+        return !this.topGroup.some((selectedItem) => this._getItemLabel(item) === this._getItemLabel(selectedItem));
+      });
```

Comment:
nit: the filteredItems array is only tested in visual tests. Removing the variable doesn't fail any unit tests.